caffeine@patapon.info/preferences/timerPage.js

```diff
@@ -0,0 +1,346 @@
+            title: _('Durations'),
+            activatable: true
+        });
+
+        let adjustSliderTimer = new Gtk.Adjustment({
+            lower: 0,
+            upper: 4,
+            step_increment: 0.1,
+            page_increment: 1,
+            value: durationIndex
+        });
+
+        this.sliderTimer = new Gtk.Scale({
+            valign: 'center',
+            hexpand: true,
+            width_request: '200px',
+            round_digits: false,
+            draw_value: false,
+            orientation: 'horizontal',
+            digits: 0,
+            adjustment: adjustSliderTimer
+        });
+        this.sliderTimer.add_mark(0, Gtk.PositionType.BOTTOM, null);
+        this.sliderTimer.add_mark(1, Gtk.PositionType.BOTTOM, null);
+        this.sliderTimer.add_mark(2, Gtk.PositionType.BOTTOM, null);
+        this.sliderTimer.add_mark(3, Gtk.PositionType.BOTTOM, null);
+        this.sliderTimer.add_mark(4, Gtk.PositionType.BOTTOM, null);
+        this.timerOptionRow.add_suffix(this.sliderTimer);
+
+        // Add elements
+        timerGroup.add(this.timerOptionRow);
+        this.add(timerGroup);
+
+        // Custom value group
+        // --------------
+        this.resetCustomTimerButton = new Gtk.Button({
+            icon_name: 'view-refresh-symbolic',
+            valign: Gtk.Align.CENTER,
+            hexpand: false,
+            vexpand: false
+        });
+
+        let customDurationGroup = new Adw.PreferencesGroup({
+            title: _('Custom durations'),
+            header_suffix: this.resetCustomTimerButton
+        });
+
+        // Custom value Adw.spinRow
+        const maxValueSecond = 359940; // = 99 Hours, 99 minutes
+        const variantDuration = this._settings.get_value(this._settingsKey.DURATION_TIMER_LIST);
+        const durationValues = variantDuration.deepUnpack();
+        this.shortTimerSelector = this.timerSpinRow(_('Short timer'),
+            60,
+            durationValues[0], // Short duration
+            60,
+            maxValueSecond - 60 * 2);
+        this.mediumTimerSelector = this.timerSpinRow(_('Medium timer'),
+            60,
+            durationValues[1], // Medium duration
+            60 * 2,
+            maxValueSecond - 60);
+        this.longTimerSelector = this.timerSpinRow(_('Long timer'),
+            60,
+            durationValues[2], // Long duration
+            60 * 3,
+            maxValueSecond);
+
+        // Enable / Disable Custom value
+        let enableCustomTimerSwitch = new Gtk.Switch({
+            valign: Gtk.Align.CENTER,
+            active: this._settings.get_boolean(this._settingsKey.USE_CUSTOM_DURATION)
+        });
+        this.enableCustomTimerRow = new Adw.ActionRow({
+            title: _('Enable custom values'),
+            subtitle: _('Select custom value for each duration'),
+            activatable_widget: enableCustomTimerSwitch
+        });
+        this.enableCustomTimerRow.add_suffix(enableCustomTimerSwitch);
```

Comment:
```suggestion
        this.enableCustomTimerRow = new Adw.SwitchRow({
            title: _('Enable custom values'),
            subtitle: _('Select custom value for each duration'),
            active: this._settings.get_boolean(this._settingsKey.USE_CUSTOM_DURATION)
        });
```
[SwitchRow](https://gjs-docs.gnome.org/adw1~1/adw.switchrow) was introduced with libadwaita 1.4 (Gnome 45). It would make sense to use it here.